MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java

```diff
@@ -231,6 +238,42 @@ public void setWebReturnToLabel(String webReturnToLabel) {
     }
 
     /**
+     * Enables Android Pay support in the {@link BuyClient}
+     * @param androidPayPublicKey The base64 encoded public key associated with Android Pay.
+     */
+    public void enableAndroidPay(String androidPayPublicKey) {
+        if (TextUtils.isEmpty(androidPayPublicKey)) {
+            throw new IllegalArgumentException("androidPayPublicKey cannot be empty");
+        }
+
+        byte[] digest;
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+            digest = messageDigest.digest(androidPayPublicKey.getBytes("UTF-8"));
+        } catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+            // Do not enable Android Pay if the hash could not be computed
```

Comment:
Should we throw Runtime exception? if not then we need to log this error, but not silently swallow.